wsi_superpixel_guided_labeling/web_client/views/vue/components/ActiveLearning/ActiveLearningContainer.vue

```diff
@@ -199,6 +205,37 @@ export default Vue.extend({
                 this.updateMapBoundsForSelection();
                 this.drawLabels();
             });
+            this.viewerWidget.on('g:drawOverlayAnnotation', (element, layer) => {
+                if (element.type === 'pixelmap') this.overlayLayer = layer;
+            });
+        },
+        updatePixelmapLayerStyle() {
+            _.forEach(this.overlayLayer.features(), (feature) => {
+                feature.style('color', (d, i) => {
+                    if (d < 0 || d >= this.categories.length) {
+                        console.warn(`No category found at index ${d} in the category map.`);
+                        return 'rgba(0, 0, 0, 0)';
+                    }
+                    const category = this.categories[d];
+                    return (i % 2 === 0) ? category.fillColor : category.strokeColor;
+                });
+            });
+            this.overlayLayer.draw();
+        },
+        saveAnnotations(imageId) {
+            // If we dont specify an image, save all images
+            const idsToSave = imageId ? [imageId] : Object.keys(this.annotationsByImageId);
+            this.backboneParent.saveLabelAnnotations(idsToSave);
+        },
+        synchronizeCategories() {
+            _.forEach(Object.values(this.annotationsByImageId), (annotations) => {
+                const superpixelElement = annotations.labels.get('annotation').elements[0];
+                if (superpixelElement) {
+                    superpixelElement.categories = JSON.parse(JSON.stringify(this.categories));
+                }
```

Comment:
This is only saving changes for the "labels" annotation. We could modify this to also change the "predictions" annotation. Both are being affected by the slider, so I think it makes the most sense to save both.

Reply:
Also I'm not sure what would happen if I save the annotations with a changed opacity and then do another round of training. Does the opacity get reset?

Reply:
> This is only saving changes for the "labels" annotation. We could modify this to also change the "predictions" annotation. Both are being affected by the slider, so I think it makes the most sense to save both.

Fixed!

Reply:
> Also I'm not sure what would happen if I save the annotations with a changed opacity and then do another round of training. Does the opacity get reset?

With the new support for the config file this value is now properly set for each round of training. This also removes the need for the "predictions" change above.